Scripts/CodeFormat/check_format.sh

```diff
@@ -38,18 +43,30 @@ finish () {
 # feature
 trap finish EXIT
 
-GIT_CLANG_FORMAT="${GIT_CLANG_FORMAT:-git-clang-format}"
-"$GIT_CLANG_FORMAT" --style=file --extensions=cc,cp,cpp,c++,cxx,cu,cuh,hh,hpp,hxx,hip,vert,frag --diff "$@" "$SOURCE_COMMIT" > "$scratch"
+CLANG_FORMAT="clang-format"
 
-# Check for no-ops
-grep '^no modified files to format$\|^clang-format did not modify any files$' \
-    "$scratch" > /dev/null && exit 0
+# Get all staged files with relevant extensions
+FILES=$(git diff --cached --name-only --diff-filter=ACM | grep -E '\.(cc|cp|cpp|c\+\+|cxx|cu|cuh|hh|hpp|hxx|hip|vert|frag)$')
+if [ -z "$FILES" ]; then
+    echo "No files to format."
+    exit 0
+fi
+
+echo "Running $CLANG_FORMAT --style=file -i $FILES"
+$CLANG_FORMAT --style=file -i $FILES
+
+# Check if any files were modified by clang-format
+MODIFIED_FILES=$(git diff --name-only --diff-filter=M $FILES)
+if [ -z "$MODIFIED_FILES" ]; then
+    echo "No formatting changes needed."
+    exit 0
+fi
```

Comment:
Why remove the use of `git-clang-format`? I don't think this new version of the script checks the modified files by the branch only

Reply:
I removed `git-clang-format` due to conflicts with `clang-format` styles and standardized on using `clang-format`. The updated formatting script now checks staged files that are added, copied, or modified (ACM). If preferred, I can revert to using `git-clang-format`, but this was also likely require updating the `linting.yml` to use `git-clang-format` as well.

Reply:
Thanks, @Beanavil. Are you using `clang-format-18` with your pre-commit hooks?

Reply:
Well, we are using the 17 version as of now, but I wanted to bump it to 18. We could also change it for now to 17 (so that we all use the same version) and later bump it. The logic to use clang-format-17 is the same as I suggested for 18, but replacing the version number.